sdk/search/search-documents/src/serviceUtils.ts

```diff
@@ -427,48 +468,87 @@ export function generatedIndexToPublicIndex(generatedIndex: GeneratedSearchIndex
     similarity: convertSimilarityToPublic(generatedIndex.similarity),
+    semanticSearch: generatedIndex.semanticSearch,
+    vectorSearch: generatedVectorSearchToPublicVectorSearch(generatedIndex.vectorSearch),
   };
 }
 
-export function generatedSearchResultToPublicSearchResult<T>(
-  results: GeneratedSearchResult[]
-): SearchResult<T>[] {
-  const returnValues: SearchResult<T>[] = results.map<SearchResult<T>>((result) => {
-    const { _score, _highlights, rerankerScore, captions, ...restProps } = result;
-    const doc: { [key: string]: any } = {
-      ...restProps,
-    };
-    const obj = {
-      score: _score,
-      highlights: _highlights,
-      rerankerScore,
-      captions,
-      document: doc,
+export function generatedVectorSearchAlgorithmConfigurationToPublicVectorSearchAlgorithmConfiguration(): undefined;
+export function generatedVectorSearchAlgorithmConfigurationToPublicVectorSearchAlgorithmConfiguration(
+  generatedAlgorithmConfiguration: GeneratedVectorSearchAlgorithmConfiguration
+): VectorSearchAlgorithmConfiguration;
+export function generatedVectorSearchAlgorithmConfigurationToPublicVectorSearchAlgorithmConfiguration(
+  generatedAlgorithmConfiguration?: GeneratedVectorSearchAlgorithmConfiguration
+): VectorSearchAlgorithmConfiguration | undefined {
+  if (!generatedAlgorithmConfiguration) {
+    return generatedAlgorithmConfiguration;
+  }
+
+  if (["hnsw", "exhaustiveKnn"].includes(generatedAlgorithmConfiguration.kind)) {
+    const algorithmConfiguration = generatedAlgorithmConfiguration as
+      | GeneratedHnswAlgorithmConfiguration
+      | GeneratedExhaustiveKnnAlgorithmConfiguration;
+    const metric = algorithmConfiguration.parameters?.metric as VectorSearchAlgorithmMetric;
+    return {
+      ...algorithmConfiguration,
+      parameters: { ...algorithmConfiguration.parameters, metric },
     };
-    return obj as unknown as SearchResult<T>;
-  });
+  }
+
+  throw Error("Unsupported algorithm configuration");
+}
+
+export function generatedVectorSearchToPublicVectorSearch(
+  vectorSearch?: GeneratedVectorSearch
+): VectorSearch | undefined {
+  if (!vectorSearch) {
+    return vectorSearch;
+  }
+
+  return {
+    ...vectorSearch,
+    algorithms: vectorSearch.algorithms?.map(
+      generatedVectorSearchAlgorithmConfigurationToPublicVectorSearchAlgorithmConfiguration
+    ),
+  };
+}
+
+export function generatedSearchResultToPublicSearchResult<
+  TModel extends object,
+  TFields extends SelectFields<TModel>
+>(results: GeneratedSearchResult[]): SearchResult<TModel, TFields>[] {
+  const returnValues: SearchResult<TModel, TFields>[] = results.map<SearchResult<TModel, TFields>>(
+    (result) => {
+      const { _score, _highlights, rerankerScore, captions, ...restProps } = result;
+      const obj = {
+        score: _score,
+        highlights: _highlights,
+        rerankerScore,
+        captions,
+        document: restProps, // todo: make this change in beta
```

Comment:
this todo item is not clear to me.

Reply:
This is just there to remind me to minimize the diff between the preview and GA packages, as I made a change that wouldn't be caught by the compiler. This will be removed when I catch the preview branch up with the breaking changes in this release.